### PR TITLE
[BEAM-5315] Partially port Avroio module

### PR DESCRIPTION
**Pull request by RobbeSneyders**
This is is part of a series of PRs with goal to make Apache Beam PY3 compatible. The proposal with the outlined approach has been documented here: https://s.apache.org/beam-python-3.

This PR partially ports the avroio module.
4 tests (2x the same 2 tests, for Avro and FastAvro) are still failing. I created [BEAM-6522](https://issues.apache.org/jira/browse/BEAM-6522) to track this.

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | --- | --- | --- | --- | ---
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_Verify/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/) </br> [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | --- | --- | --- | ---

### CODE REVIEW

**Comment by RobbeSneyders** [comment]
Tests currently fail due to `tox.ini` changes. When #7583 gets merged, this should not be an issue anymore.

**Comment by tvalentyn** [comment]
retest this please

**Comment by aaltay** [comment]
hey @RobbeSneyders could you resolve the conflict please?

File: sdks/python/tox.ini [hunk 1]
```diff
@@ -57,17 +57,13 @@ commands =
...
 commands =
   python --version
   pip --version
   {toxinidir}/scripts/run_tox_cleanup.sh
-  python setup.py nosetests --tests {[testenv:py3]modules}
+  python apache_beam/examples/complete/autocomplete_test.py
```
[the remark this answers is not in the document]
Comment (RobbeSneyders) [reply sdks/python/tox.ini]: This test is executed in all other tox environments. I don't know why it is added separately, so I also don't know why we shouldn't do it in Python 3.

File: sdks/python/tox.ini [hunk 2]
```diff
@@ -57,17 +57,13 @@ commands =
...
+  python setup.py nosetests --ignore-files vcfio_test\.py
```
[the remark this answers is not in the document]
Comment (RobbeSneyders) [reply sdks/python/tox.ini]: Ok, sounds good.

**Comment by RobbeSneyders** [comment]
PTAL

File: sdks/python/tox.ini (hunk 1 above)
[the remark this answers is not in the document]
Comment (tvalentyn) [reply sdks/python/tox.ini]: As per https://github.com/apache/beam/commit/245facdbc166298cdb96521165a7514653fa8e57 we  try to execute one test outside of the test suite to make sure nothing fails if we do so.

**Comment by tvalentyn** [comment]
LGTM. Thank you.

**Comment by charlesccychen** [comment]
Thanks, this LGTM.

**Pull request merged**